pkg/estimator/client/service.go

```diff
@@ -41,7 +41,7 @@ func resolveCluster(kubeClient kubernetes.Interface, namespace, id string, port
 			 * But the Service resource is defined in Host Kubernetes Cluster. So we cannot get its content here.
 			 * The best thing we can do is just glue host:port together, and try to connect to it.
 			 */
-			return net.JoinHostPort(fmt.Sprintf("%s.%s.svc.cluster.local", id, namespace), fmt.Sprintf("%d", port)), nil
```

Comment:
Is this compatible with the old version?

Reply:
you mean change `%s.%s.svc.cluster.local` to `%s.%s`?

Reply:
can this provide any improvement?

Reply:
The purpose of this modification is that some environments have a default DNS suffix that is not `svc.cluster.local`. By completely removing that segment, the estimator can run properly in these environments.

Reply:
However, there is now a certificate issue. The old certificate does not include `*.karmada-system`, which can cause upgrade problems. I am currently trying to resolve this.

Reply:
refer to [Service DNS](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#services),
>  with a name of the form `my-svc.my-namespace.svc.cluster-domain.example`

looks like the `svc` part of DNS is not configurable. If so, we can use `ServiceName.serviceNamspace.svc` as the address of the gRPCConnection, without worrying about the incompatibility issues of certificates, because the current karmada certificate already has the host `*.karmada-system.svc`. 

WDYT @chaunceyjiang @whitewindmills

Reply:
seems that such domain `ServiceName.serviceNamspace.svc` is not be supported by windows nodes.
https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#dns-windows

Reply:
I believe we can first use `*.karmada-system.svc.`. Then in the certificate, add `*.karmada-system.`. In the next version, use `*.karmada-system.`.

Reply:
> seems that such domain ServiceName.serviceNamspace.svc is not be supported by windows nodes.
https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#dns-windows

@whitewindmills Yeah, looks like windows nodes only support domain `ServiceName`. So, changing it to `*.karmada-system.svc` will result in the loss of support for Windows nodes while adding support for Linux custom DNS suffixes. I am not certain if this trade-off is entirely acceptable.

I currently have two proposals:
- Change `ServiceName.serviceNamespace.svc.cluster.local` to `ServiceName.serviceNamespace.svc`.
- Implement a mechanism in gRPC for polling multiple addresses, using the first successfully dialed address from either `ServiceName.serviceNamespace.svc.cluster.local` or `ServiceName.serviceNamespace.svc` to establish the gRPC connection. 

The benefit of the second proposal is that it maintains compatibility with Windows nodes.

WDYT? @chaunceyjiang @whitewindmills

Reply:
kindly ping @chaunceyjiang @whitewindmills

Reply:
I see a lot of `*.svc.cluster.local` usage in karmada-operator and hack/*.sh, I'm just curious why there is no negative impact in such an environment.

Reply:
> I see a lot of *.svc.cluster.local usage in karmada-operator and hack/*.sh

For `hack/*.sh`, `*.svc.cluster.local` and `*.svc` appear together, which means that `*.svc` can support custom DNS suffixes.

Reply:
However, for `karmada-operator`, I haven't studied it in detail yet. From my understanding, `karmada-operator` should not directly use `*.svc.cluster.local`, because `karmada-operator` supports custom DNS suffixes.

https://github.com/karmada-io/karmada/blob/bf83dfe87c9bb222047197a2d2e91e7fa8464f5e/operator/pkg/apis/operator/v1alpha1/type.go#L173-L178

Reply:
however, this is just one of them. I have seen such usage mainly concentrated on the deployment of etcd.
https://github.com/karmada-io/karmada/blob/6bbbf9d8aeca49ae1f412da117ca17f995c42077/operator/pkg/controlplane/etcd/etcd.go#L54-L58

Reply:
Yes, for `karmada-operator`, there needs to be better support for the API `DNSDomain`.

Reply:
> however, this is just one of them. I have seen such usage mainly concentrated on the deployment of etcd.

@whitewindmills  The `karmadas.operator.karmada.io` API provides a field for custom DNSDomain, but it is not actually used. I will track this through an issue. Besides this, do you have any review comments on this pull request?